### ISSUE TITLE
EntLib provider doesn't support nested or mapped diagnostics contexts

### ISSUE THREAD

**Issue by damianh** [damianh/LibLog#69]
https://msdn.microsoft.com/en-us/library/ff648083.aspx
http://www.codeproject.com/Articles/9545/Get-Logging-with-the-Enterprise-Library

Would love a contrib here (hint hint @mihasic :wink: )

**Comment by damianh** [comment]
I'm thinking abbout removing support for EntLib for v5.0. Would be interested to hear of any users who would like support to continue.

**Comment by RaviKaruturi** [comment]
Sorry, to ask here.. is there any example to show how to use EntLib and LibLog together. We are having a hard time finding an example. Any help would be greatly appreciated.
Btw, What did you decide about supporting EntLib for v5.0?

**Comment by mihasic** [comment]
AFAIK, EntLib support is going to be dropped (in the code I still see the EntLibLogProvider).
The easiest way to integrate any non-supported log provider is by implementing ILogProvider and ILog. Then registering: LogProvider.SetCurrentLogProvider(yourLogProvider).

By default, it should pick up the log provider based on assemblies loaded into current AppDomain.

**Comment by RaviKaruturi** [comment]
Thanks for replay @mihasic 
What is the reason behind  removing support for EntLib?
I have figured a way to enable EntLib..

Put the following code in startup.cs

using Microsoft.Practices.EnterpriseLibrary.Common.Configuration;
using Microsoft.Practices.EnterpriseLibrary.Data;
using Microsoft.Practices.EnterpriseLibrary.Logging;

DatabaseProviderFactory ELfactory = new DatabaseProviderFactory(new SystemConfigurationSource(false).GetSection);
DatabaseFactory.SetDatabaseProviderFactory(ELfactory, false);
LogWriterFactory logWriterFactory = new LogWriterFactory();
LogWriter logWriter = logWriterFactory.Create();
Logger.SetLogWriter(logWriter);

configure Database in config file.

Note: While experimenting you should use only one logging framework in project. If more than one logging frameworks are used then once LibLog detects one logging framework it will not look for other.

**Comment by mihasic** [comment]
@RaviKaruturi, the reason is that EntLib does not support nested context (the title of this issue) out of the box and it would take quite some effort to implement.
Initially I've added EntLib stuff just for fun and because we've used EntLib in one of the projects. Even though, in the actual project I've had to write custom log provider for internal specifics. And now the project switched to Serilog.

If you or somebody would be interested on keeping EntLib support up to date, feel free to submit pull request.
I think @damianh does not really want to spend a lot of time and effort on the one. Same for myself.

> Note: While experimenting you should use only one logging framework in project. > If more than one logging frameworks are used then once LibLog detects one logging framework it will not look for other.
Yep, that's how default detection mechanism works if you do not override it.
